chain-prometheus-exporter/scripts/build.Dockerfile

```diff
@@ -14,3 +14,10 @@ RUN apt-get update && \
     && rm -rf /var/lib/apt/lists/*
 COPY --from=build /build/chain-prometheus-exporter/target/release/chain-prometheus-exporter /usr/local/bin/
 
+WORKDIR /app
+RUN useradd -U -d /app appuser
+RUN chown appuser:appuser /app
+USER appuser
```

Comment:
THis is all very ad-hoc and undocumented. moreover you have not documented this in the README either.

I am also not sure this is really the option we want to go with. Shouldn't this be configured when you are running the service instead? The equivalent of specifying the `user`?

```
podman run --entrypoint chain-prometheus-exporter --user 1000:1000 -e CHAIN_PROMETHEUS_EXPORTER_ACCOUNTS='foo:3kBx2h5Y2veb4hZgAJWPrr8RyQESKm5TjzF3ti1QQ4VSYLwK1G' -e CHAIN_PROMETHEUS_EXPORTER_CONCORDIUM_NODE='http://node.testnet.concordium.com:20000' chain-prometheus-exporter:latest
```

It is ultimately in the interest of the runner to make sure they are safe.

Reply:
Right now we have two users: `root` and `appuser`. All I am doing in this PR is to say we default to `appuser` because we know that is the more secure. Why would you want to default to `root`?

I am confused about what you mean by undocumented. Would you mention that we are creating an `appuser` in the docker image in the README?

Reply:
If you would rather use root for some reason you can just specify `--user 0:0`. But one should never use root, and in general one should try and build images which does not share uid with the host.

I get that you can change it, but I do not see any reason why not to help the caller.

Reply:
You are not really helping the caller. You are setting some defaults that are ad-hoc and undocumented.

The image already has a number of users, for example `bin` user . It has more than `root`. It is based on debian:buster.

Using named users just seems to produce noise rather than value. Why not just use the UID directly. I don't see what creating a home directory brings for example other than seeming complexity and opportunity for failure.

The image as it is is purely a packaging of a process. I don't see value in making it more complicated than that. Since the binary inside the image is purely static (or technically could be, there is libc dependency now) using a docker image is already bringing in a lot of needless complexity.